client/src/views/DeviceJoinOrganizationModal.vue

```diff
@@ -40,8 +208,261 @@ function parseInvitationLink(link: string): ClaimDeviceLink {
 }
 
 const claimDeviceLink = parseInvitationLink(props.invitationLink);
-const waitingForHost = ref(true);
+const waitingForHost = ref(false);
+
+interface Title {
+  title: string,
+  subtitle?: string,
+}
+
+const titles = new Map<DeviceJoinOrganizationStep, Title>([[
+  DeviceJoinOrganizationStep.Information, {
+    title: t('ClaimDeviceModal.titles.claimDevice'),
+  }], [
+  DeviceJoinOrganizationStep.GetHostSasCode, {
+    title: t('ClaimDeviceModal.titles.getCode'),
+    subtitle: t('ClaimDeviceModal.subtitles.getCode'),
+  }], [
+  DeviceJoinOrganizationStep.ProvideGuestCode, {
+    title: t('ClaimDeviceModal.titles.provideCode'),
+    subtitle: t('ClaimDeviceModal.subtitles.provideCode'),
+  }], [
+  DeviceJoinOrganizationStep.Password, {
+    title: t('ClaimDeviceModal.titles.password'), subtitle: t('ClaimDeviceModal.subtitles.password'),
+  }], [
+  DeviceJoinOrganizationStep.Finish, {
+    title: t('ClaimDeviceModal.titles.done', {org: ''}),
+  }],
+]);
+
+function selectHostSas(_code: string | null): void {
+  nextStep();
+}
+
+function getNextButtonText(): string {
+  if (pageStep.value === DeviceJoinOrganizationStep.Information) {
+    return t('ClaimDeviceModal.buttons.understand');
+  } else if (pageStep.value === DeviceJoinOrganizationStep.Password) {
+    return t('ClaimDeviceModal.buttons.password');
+  } else if (pageStep.value === DeviceJoinOrganizationStep.Finish) {
+    return t('ClaimDeviceModal.buttons.login');
+  }
+
+  return '';
+}
+
+const nextButtonIsVisible = computed(() => {
+  return (
+    pageStep.value === DeviceJoinOrganizationStep.Information
+    || pageStep.value === DeviceJoinOrganizationStep.Password
+    || pageStep.value === DeviceJoinOrganizationStep.Finish
+  );
+});
+
+const canGoForward = computed(() => {
+  const currentPage = getCurrentPage();
+
+  if (
+    currentPage && currentPage.value
+    && (pageStep.value === DeviceJoinOrganizationStep.Information
+    || pageStep.value === DeviceJoinOrganizationStep.Password)
+  ) {
+    return currentPage.value.areFieldsCorrect();
+  }
+  return true;
+});
+
+function getCurrentPage(): Ref<any> {
```

Comment:
This is only used by `canGoForward` and it seems most pages are not used, it should probably be removed as well as every page reference except `passwordPage`.

Reply:
I removed the used references and the `canGoForward` function